core/drivers/stm32_rtc.c

```diff
@@ -687,7 +1059,32 @@ static TEE_Result stm32_rtc_probe(const void *fdt, int node,
 		clk_disable(rtc_dev.pclk);
 	}
 
-	rtc_register(&stm32_rtc);
+	if (IS_ENABLED(CFG_STM32_RTC_HIGH_ACCURACY)) {
+		res = stm32_rtc_init();
```

Comment:
for consistency, could you rename the function to `stm32_rtc_init_high_accuracy()` or like?